### PR TITLE
 HTBHF-971 Add common event logger to java-common.

### PR DESCRIPTION
**Pull request by mikeSmith08**

### CODE REVIEW

File: common_logging/src/main/java/uk/gov/dhsc/htbhf/logging/Event.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@Builder
+public class Event {
+    private EventType eventType;
+    private LocalDateTime timestamp;
+    private Map<String, Object> eventMetadata;
+
+    //This is here to make sure that the field "eventMetadata" isn't shown in the logs
+    @JsonAnyGetter
+    public Map<String, Object> getEventMetadata() {
+        return eventMetadata;
```
Comment (YetAnotherMatt) [inline common_logging/src/main/java/uk/gov/dhsc/htbhf/logging/Event.java]: Does the use of a map here mean that the order of fields logged as json will be indeterminate? If so, can we `return new TreeMap<String, Object>(eventMetadata)` instead?

Comment (YetAnotherMatt) [reply common_logging/src/main/java/uk/gov/dhsc/htbhf/logging/Event.java]: what happens if we provide a null map?

File: common_logging/src/main/java/uk/gov/dhsc/htbhf/logging/LoggingConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Configuration that can be used to setup the {@link EventLogger}. The {@link ObjectMapper} created in this configuration
+ * is specifically created as a private instance so that it doesn't clash with any ObjectMapper is the application context
```
Comment (gethyn1) [inline common_logging/src/main/java/uk/gov/dhsc/htbhf/logging/LoggingConfiguration.java]: typo (is -> in):
```
... ObjectMapper is the application ...
```

File: common_logging/src/main/java/uk/gov/dhsc/htbhf/logging/Event.java (hunk 1 above)
Comment (mikeSmith08) [reply common_logging/src/main/java/uk/gov/dhsc/htbhf/logging/Event.java]: If we provide a null Map then no metadata is shown in the JSON at all, it is handled nicely by default.

File: common_logging/src/main/java/uk/gov/dhsc/htbhf/logging/Event.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+@Builder
+public class Event {
+    private EventType eventType;
+    private LocalDateTime timestamp;
+    private Map<String, Object> eventMetadata;
+
+    //This is here to make sure that the field "eventMetadata" isn't shown in the logs
+    @JsonAnyGetter
+    public Map<String, Object> getEventMetadata() {
+        Map<String, Object> metadata = new TreeMap<>();
```
Comment (YetAnotherMatt) [inline common_logging/src/main/java/uk/gov/dhsc/htbhf/logging/Event.java]: how about: `return eventMetadata == null ? Collections.emptyMap() : new TreeMap<>(eventMetadata)`?

**Review by YetAnotherMatt — DISMISSED** [review]

**Review by YetAnotherMatt — APPROVED** [review]

**Pull request merged**